sstables_loader.cc

```diff
@@ -148,12 +148,74 @@ class sstable_streamer {
     future<> stream_sstable_mutations(const dht::partition_range&, std::vector<sstables::shared_sstable>, bool primary_replica_only);
 };
 
+class tablet_sstable_streamer : public sstable_streamer {
+    const locator::tablet_map& _tablet_map;
+public:
+    tablet_sstable_streamer(netw::messaging_service& ms, replica::database& db, ::table_id table_id, std::vector<sstables::shared_sstable> sstables)
+        : sstable_streamer(ms, db, table_id, std::move(sstables))
+        , _tablet_map(_erm->get_token_metadata().tablets().get_tablet_map(table_id)) {
+    }
+
+    virtual future<> stream(bool primary_replica_only) override;
+private:
+    future<> stream_fully_contained_sstables(const dht::partition_range& pr, std::vector<sstables::shared_sstable> sstables,
+                                             bool primary_replica_only) {
+        // FIXME: fully contained sstables can be optimized.
```

Comment:
Let's add a little bit more details for the optimization. E.g, we can stream the whole file with file stream in case we want to stream the whole sst file.